src/main/scala/freeeventsourcing/syntax/ProcessSyntax.scala

```diff
@@ -0,0 +1,218 @@
+
+  import Builders._
+
+  /** Await an event using a selector. The event must be from this bounded context. */
+  def await[S <: WithEventType: EventSelector: ValidSelector](selector: S) =
+    lift(AwaitEvent[BC, S](selector))
+
+  /** Send commands to a specific aggregate. */
+  def on[Id, A](aggregate: Id)(implicit afi: AggregateFromId[Id, BC#Aggregates]) = {
+    val aggregateType = afi.aggregate(boundedContext.aggregates)
+    new OnAggregateBuilder[afi.Out](aggregateType, aggregate)
+  }
+
+  /** Await events from a specific aggregate. */
+  def from[Id, A](aggregate: Id)(implicit afi: AggregateFromId[Id, BC#Aggregates]) = {
+    val aggregateType = afi.aggregate(boundedContext.aggregates)
+    new FromAggregateBuilder[afi.Out](aggregateType, aggregate)
+  }
+
+  /** Wait until the specified date/time. If the time has already passed it will still be called. */
+  def waitUntil(when: Instant) =
+    lift[Unit](WaitUntil[BC](when))
+
+  /** Wait for multiple events and run the path of the first event. */
+  def firstOf[Paths <: HList](b: FirstOfBuilder[HNil] ⇒ FirstOfBuilder[Paths])(implicit switch: Switch[Paths]): ProcessMonad[switch.Result] = {
+    val paths = b(new FirstOfBuilder(HNil)).collect
+    val alternatives = switch.alternatives(paths)
+    lift[alternatives.Events](FirstOf[BC, alternatives.type](alternatives))
+      .flatMap(e ⇒ switch.effectFor(paths)(e))
+  }
+
+  /** Wait for multiple events and run the path of the first event. */
+  def firstOfUnified[Paths <: HList, R <: Coproduct](b: FirstOfBuilder[HNil] ⇒ FirstOfBuilder[Paths])(implicit s: Switch.Aux[Paths, R], u: Unifier[R]) =
+    firstOf(b).map(r ⇒ u(r))
+
+  /** Terminate this process instance. */
+  def terminate = lift[Unit](End[BC]())
+
+  /** Does nothing. */
+  def noop = Monad[ProcessMonad].pure(())
+
+  /////////////////////////////////////////////////////
+
+  private def lift[A](action: ProcessAction[BC, A]) = Free.liftF[ProcessAction[BC, ?], A](action)
+
+  /** Internas used by the above functions. Need to be public but are not meant to be used directly. */
+  object Builders {
+    /** Helper class for on. */
+    final class OnAggregateBuilder[A <: Aggregate] private[ProcessSyntax] (aggregateType: A, aggregate: A#Id) {
+      /** Execute a command on the aggregate. */
+      def execute[R <: Coproduct](command: A#Command)(
+        catches: CommandErrorHandler.EmptyBuilder[command.Error] ⇒ CommandErrorHandler.Builder[command.Error, R]
+      )(implicit ev: ValidAggregate[BC, A], ev2: AllErrorsHandled[R]) = {
+        val builder = CommandErrorHandler.builder[command.Error]
+        val errorHandler = catches(builder).errorHandler
+        lift[Unit](Execute[BC, A, command.type](aggregateType, aggregate, command, errorHandler))
+      }
+    }
+
+    /** Helper class for from. */
+    final class FromAggregateBuilder[A <: Aggregate] private[ProcessSyntax] (aggregateType: A, aggregate: A#Id) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class name does not match the regular expression '"[A-Z][A-Za-z]*"'](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2571398663/issues/source?bid=3298199&fileBranchId=3298202#l82)